cli/Makefile

```
@@ -17,15 +17,33 @@ GO_FLAGS += -trimpath
 
 CLI_DIR = $(shell pwd)
 
+# allow opting in to the rust codepaths
+GO_TAG ?= go
+
 GO_FILES = $(shell find . -name "*.go")
 SRC_FILES = $(shell find . -name "*.go" | grep -v "_test.go")
 GENERATED_FILES = internal/turbodprotocol/turbod.pb.go internal/turbodprotocol/turbod_grpc.pb.go
 
 turbo: go-turbo$(EXT)
 	cargo build --manifest-path ../crates/turborepo/Cargo.toml
 
-go-turbo$(EXT): $(GENERATED_FILES) $(SRC_FILES) go.mod
-	CGO_ENABLED=1 go build $(GO_FLAGS) -o go-turbo$(EXT) ./cmd/turbo
+go-turbo$(EXT): $(GENERATED_FILES) $(SRC_FILES) go.mod turborepo-ffi-install
+	CGO_ENABLED=1 go build $(GO_FLAGS) -tags $(GO_TAG) -o go-turbo$(EXT) ./cmd/turbo
+
+.PHONY: turborepo-ffi-install
+turborepo-ffi-install: turborepo-ffi turborepo-ffi-proto
+	cp -r ../crates/turborepo-ffi/bindings.h \
+	      ../crates/turborepo-ffi/ffi/proto \
+	      ../crates/turborepo-ffi/target/release/libturborepo_ffi.a \
+	      ./internal/ffi
+
+.PHONY: turborepo-ffi
+turborepo-ffi:
+	cd ../crates/turborepo-ffi && CARGO_TARGET_DIR=./target cargo build --release
+
+.PHONY: turborepo-ffi-proto
+turborepo-ffi-proto:
+	cd ../crates/turborepo-ffi && protoc --go_out=. messages.proto
```

Mild preference for this, which is untested but I think it works:

```suggestion
	cd ../crates/turborepo-ffi && protoc --go_out=$(CLI_DIR)/ffi/proto messages.proto
```